File: app/src/main/java/torille/fi/lurkforreddit/data/models/jsonResponses/StreamableVideo.java
package torille.fi.lurkforreddit.data.models.jsonResponses;

import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

/**
 * Class that holds streamable response
 */
@AutoValue
public abstract class StreamableVideo {

    @SerializedName("title")
    public abstract String title();

    @SerializedName("files")
    public abstract Videos videos();


    public static TypeAdapter<StreamableVideo> typeAdapter(Gson gson) {
        return new AutoValue_StreamableVideo.GsonTypeAdapter(gson);

    }

    @AutoValue
    public abstract static class Videos {

        @Nullable
        @SerializedName("mp4-mobile")
        public abstract ImageResolution mobileVideo();

        @SerializedName("mp4")
        public abstract ImageResolution video();

        public static TypeAdapter<Videos> typeAdapter(Gson gson) {
            return new AutoValue_StreamableVideo_Videos.GsonTypeAdapter(gson);

        }
    }

}

File: app/src/main/java/torille/fi/lurkforreddit/data/models/jsonResponses/StreamableVideo.kt
package torille.fi.lurkforreddit.data.models.jsonResponses

import com.google.gson.annotations.SerializedName

/**
 * Class that holds streamable response
 */
data class Videos(
        @SerializedName("mp4-mobile")
        val mobileVideo: ImageResolution?,
        @SerializedName("mp4")
        val video: ImageResolution
)

data class StreamableVideo(
        @SerializedName("title")
        val title: String,
        @SerializedName("files")
        val videos: Videos
)